### PR TITLE
chore: 15120 renames and reorg

### LINKED ISSUE #15120: Pairings Api Name refactors
Given the recent confusion in trying to explain the concepts, i came up with a new organization and naming for some of the pieces.

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
**Description:**
Given some of the challenges and confusion related to naming, I am providing a different naming strategy.

image

Related issue(s):

Fixes https://github.com/hashgraph/hedera-services/issues/15120

**Notes for reviewer:**
Binary files are provided until compiling support with gradle for rust is completed.

### CODE REVIEW

**Comment by lpetrovic05** [comment]
the diff looks like its a lot more than a rename

**Comment by mxtartaglia-sl** [comment]
@lpetrovic05 this depends on the previous pr, i changed the baseline to reflect that.
Thanks

File: cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java [hunk 1]
```diff
@@ -6,9 +8,9 @@
     requires com.hedera.cryptography.pairings.api;
     requires jdk.jshell;
 
-    uses com.hedera.cryptography.pairings.spi.BilinearPairingProvider;
+    uses PairingFriendlyCurveProvider;
```
Comment (edward-swirldslabs) [inline cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java:11]: Is it better to have the full package path on this line, or in the import statement at the top of the file?  It seems to be standard practice to not use import statements in module-info?

**Review by edward-swirldslabs — DISMISSED** [review]

File: cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java (hunk 1 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java:11]: This is IntelliJ doing when refactoring. I keep coming back and changing it back to the complete reference. 
I think better let's go with what intellij does by default.

Comment (edward-swirldslabs) [reply cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java:11]: ah, I wonder if that is a configurable setting.   Agreed to just adopt what intellij does or have spotless apply fix it if it is a coding style error.

**Review by lpetrovic05 — APPROVED** [review]

File: cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java (hunk 1 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-altbn128/src/main/java/module-info.java:11]: Resolving this, i'll take a look later if it can be modified.

**Pull request merged**